File: core/src/no/elg/infiniteBootleg/console/StdConsole.java
package no.elg.infiniteBootleg.console;

import com.strongjoshua.console.HeadlessConsole;
import no.elg.infiniteBootleg.Main;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

/**
 * A console that reads input from standard in
 */
public class StdConsole extends HeadlessConsole implements Runnable {

    private BufferedReader in;
    private Thread thread;
    private boolean running;

    public StdConsole() {
        running = true;

        thread = new Thread(this, "Headless Console Reader Thread");
        thread.setDaemon(true);
        thread.start();
    }

    @Override
    public void run() {
        in = new BufferedReader(new InputStreamReader(System.in));
        while (running) {
            try {
                String read = in.readLine();
                Main.SCHEDULER.executeSync(() -> execCommand(read));
            } catch (IOException e) {
                e.printStackTrace();
            }
        }
    }

    @Override
    public void dispose() {
        super.dispose();
        running = false;
    }
}

File: core/src/main/kotlin/no/elg/infiniteBootleg/core/console/consoles/StdConsole.kt
package no.elg.infiniteBootleg.core.console.consoles

import com.strongjoshua.console.HeadlessConsole
import no.elg.infiniteBootleg.core.console.HelpfulConsoleHelpUtil

/** A console that reads input from standard in  */
class StdConsole : HeadlessConsole() {
  override fun printHelp(command: String) {
    HelpfulConsoleHelpUtil.printHelp(this, exec, command)
  }

  override fun printCommands() {
    HelpfulConsoleHelpUtil.printCommands(this, exec)
  }

  override fun isVisible(): Boolean = true
}
